packages/shared/lib/core/network/classes/stardust-network.class.ts

```diff
@@ -13,14 +13,14 @@ import { IscpChain } from './iscp-chain.class'
 
 export class StardustNetwork implements INetwork {
     private readonly _metadata: NetworkMetadata
-    private readonly _chains: IChain[]
+    private readonly _chains: (IChain | undefined)[]
 
     constructor(metadata: NetworkMetadata, chainConfigurations: ChainConfiguration[]) {
         this._metadata = metadata
         this._chains = this.constructChains(chainConfigurations ?? [])
     }
 
-    private constructChains(chainConfigurations: ChainConfiguration[]): IChain[] {
+    private constructChains(chainConfigurations: ChainConfiguration[]): (IChain | undefined)[] {
```

Comment:
i think we shouldnt allow `undefined` to be in the chains. After the `.map` we should filter for all non-undefined elements